apps/orgs/utils.py

```diff
@@ -122,9 +128,10 @@ def tmp_to_builtin_org(system=0, default=0):
     else:
         raise ValueError("Must set system or default")
     ori_org = get_current_org()
-    set_current_org(org_id)
-    yield
-    if ori_org is not None:
+    try:
+        set_current_org(org_id)
+        yield
+    finally:
         set_current_org(ori_org)
 
 
```

Comment:
I have reviewed the provided code snippets and could not find any errors or unnecessary features. However, I suggest optimizing the `get_current_org_id_for_serializer()` function slightly to make it more efficient.

The current implementation seems correct but can be made a bit faster by removing redundant checks using `.not_` before checking that `org` exists:

```python
@check_not_none(lambda: org)
def get_current_org_id_for_serializer(self):
```
This modification makes sure each call to this method has its own independent check (`check_not_none`) without creating multiple copies of that logic in every instance of your decorator.